client.go

```diff
@@ -62,7 +62,15 @@ func HTTPClient(h *http.Client) ClientParam {
 // BasicAuth sets username and password for HTTP client
 func BasicAuth(username, password string) ClientParam {
 	return func(c *Client) error {
-		c.basicAuth = &basicAuth{username: username, password: password}
+		c.auth = &basicAuth{username: username, password: password}
+		return nil
+	}
+}
+
+// BearerAuth sets username and token for HTTP client
```

Comment:
Does not set the username - only token.
That's my the problem with comments (or maybe certain comments?) - they become obsolete pretty much immediately and otherwise require pretty careful formulation (avoiding too many implementation details and such)